presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/operator/aggregation/UnionAgg.java

```diff
@@ -0,0 +1,92 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.geospatial.operator.aggregation;
+
+import com.esri.core.geometry.ogc.OGCGeometry;
+import com.facebook.presto.geospatial.serde.GeometrySerde;
+import com.facebook.presto.plugin.geospatial.GeoFunctions;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.function.AggregationFunction;
+import com.facebook.presto.spi.function.AggregationState;
+import com.facebook.presto.spi.function.CombineFunction;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.InputFunction;
+import com.facebook.presto.spi.function.OutputFunction;
+import com.facebook.presto.spi.function.SqlType;
+import io.airlift.slice.Slice;
+
+import java.util.EnumSet;
+
+import static com.facebook.presto.geospatial.GeometryType.GEOMETRY_COLLECTION;
+import static com.facebook.presto.geospatial.GeometryType.LINE_STRING;
+import static com.facebook.presto.geospatial.GeometryType.MULTI_LINE_STRING;
+import static com.facebook.presto.geospatial.GeometryType.MULTI_POINT;
+import static com.facebook.presto.geospatial.GeometryType.MULTI_POLYGON;
+import static com.facebook.presto.geospatial.GeometryType.POINT;
+import static com.facebook.presto.geospatial.GeometryType.POLYGON;
+import static com.facebook.presto.plugin.geospatial.GeometryType.GEOMETRY;
+import static com.facebook.presto.plugin.geospatial.GeometryType.GEOMETRY_TYPE_NAME;
+
+/**
+ * Aggregate form of st_union which takes a set of geometries and unions them into a single geometry, resulting in no intersecting
+ * regions.  The output may be a MULTI*, single geometry, or geometry collection.
+ */
+@Description("Returns a geometry that represents the point set union of the input geometries.")
+@AggregationFunction("union_agg")
+public class UnionAgg
+{
+    private UnionAgg() {}
+
+    /**
+     * The input function, which takes a Slice, deserializes it, and unions it into the {@link GeometryState}
+     * @param state the state object which represents an aggregate geometry
+     * @param slice the slice input
+     */
+    @InputFunction
+    public static void input(@AggregationState final GeometryState state, @SqlType(GEOMETRY_TYPE_NAME) final Slice slice)
+    {
+        OGCGeometry geometry = GeometrySerde.deserialize(slice);
+        GeoFunctions.validateType("union_agg", geometry, EnumSet.of(POINT, MULTI_POINT, LINE_STRING, MULTI_LINE_STRING, POLYGON, MULTI_POLYGON, GEOMETRY_COLLECTION));
+        state.union(geometry);
```

Comment:
perhaps, check if geometry is empty and skip `union` call; this way you won't need to workaround an issue the `estimateMemorySize`

Reply:
Good idea.  In case you didn't think about this: there is information loss when one unions `POINT EMPTY`, `POINT LINESTRING` in this case (it becomes either `POINT EMPTY` or `null` depending on how it's coded, vs. `GEOMETRYCOLLECTION EMPTY`), but I think it's acceptable and not a big deal.

Reply:
@tdcmeehan Tim, I think it is OK to lose type info when union-ing two empty geometries of different types, but I think the result shouldn't be `null`. In which case would we get a `null`?

Reply:
Wouldn't we still need some sort of workaround in the case of unioning two `LINESTRING EMPTY`?  If we set the state to the first geometry, but avoid unioning the second, we're back where we started.  If we avoid setting it to begin with (because it's empty), then we get `null`.

Reply:
@tdcmeehan Tim, you are right. We could do something like: 

```
if (getGeometry() == null) {
   setGeometry(geometry);
}
else if (!geometry.isEmpty()) {
  setGeometry(getGeometry().union(geometry));
}
```

and then estimate memory size for empty geometries as `24` (the base size of an `OGCGeometry` object; until the next release of the library).